### ISSUE TITLE
Modules opening firewall ports by default

### ISSUE THREAD

**Issue by lunik1** [NixOS/nixpkgs#149889]
## Issue description
Certain NixOS modules open firewall ports by default. [According to the Nixpkgs manual](https://nixos.org/manual/nixpkgs/stable/#reviewing-contributions-new-modules) this is not desired behaviour, with an exception for sshd https://github.com/NixOS/nixpkgs/pull/75454#issuecomment-564573908.

This could break existing setups, so I would be interested to here from maintainers what the best mitigations would be e.g.
- Recording the changed default in release notes
- Warning at evaluation if `open{Ports|Firewall}` is not explicitly set for these services
- Gating the new default behaviour behind a `stateVersion` check

Known offending modules:
 - [ ] services/networking/unifi.nix
 - [ ] services/video/unifi-video.nix
 - [ ] services/networking/avahi-daemon.nix
 - [ ] services/cluster/hadoop/hdfs.nix
 - [ ] services/audio/snapserver.nix

## Notify Maintainers
@erictapen @pennae (unifi)
@rsynnest (unifi-video)
@tobim (snapserver)

**Comment by erictapen** [comment]
I think the evaluation warning makes the most sense, but it should mention some kind of deprecation window, so it can be removed in a year or so. Would be happy to review such a PR for `services/networking/unifi.nix`, but I'm afraid I have no time currently to implement it myself.

**Comment by tobim** [comment]
I agree with a deprecation warning, but we should also put it into the release notes. I'll try to fix snapcast by the end of the year.

**Comment by lunik1** [comment]
Pings for the remaining modules based off git blame, sorry if it should have been somebody else!
@WilliButz (avahi) @illustris (hdfs).

**Comment by lunik1** [comment]
Based on discussions here and on matrix we are leaning towards warning in 22.05 if not explicitly set and changing the default to false in 22.11.

**Comment by bb2020** [comment]
> Certain NixOS modules open firewall ports by default. [According to the Nixpkgs manual](https://nixos.org/manual/nixpkgs/stable/#reviewing-contributions-new-modules) this is not desired behaviour, with an exception for sshd [#75454 (comment)](https://github.com/NixOS/nixpkgs/pull/75454#issuecomment-564573908).

Also see #81490.

**Comment by Pandapip1** [comment]
The stated justification for `openFirewall` not being enabled by default is not trampling on other modules (it's under the "Ensure that the module respect[sic] other modules functionality"). Is there a reason that this issue has been tagged with the `1.severity: security` label when there is no security justification provided for the rule in the first place?
